static/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -182,18 +190,48 @@ class SpanTree extends React.Component<PropType> {
 
         const {span} = payload;
 
-        const key = getSpanID(span, `span-${index}`);
+        let spanNumber = acc.spanNumber;
 
+        const key = getSpanID(span, `span-${spanNumber}`);
         const isLast = payload.isLastSibling;
         const isRoot = type === 'root_span';
         const spanBarColour: string = pickBarColour(getSpanOperation(span));
-        const spanNumber = index + 1;
         const numOfSpanChildren = payload.numOfSpanChildren;
         const treeDepth = payload.treeDepth;
         const continuingTreeDepths = payload.continuingTreeDepths;
 
         acc.numOfFilteredSpansAbove = 0;
         acc.numOfSpansOutOfViewAbove = 0;
+
+        const hasCollapsedSpanGroup: boolean =
+          (payload.type === 'span' || payload.type === 'root_span') &&
+          isCollapsedSpanGroup({
+            spanGrouping: payload.spanGrouping,
+            showSpanGroup: payload.showSpanGroup,
+            toggleSpanGroup: payload.toggleSpanGroup,
+          });
+
+        if (hasCollapsedSpanGroup) {
+          assert(payload.type === 'span' || payload.type === 'root_span');
```

Comment:
Hey Alberto, is the `assert` here necessary? `hasCollapsedSpanGroup` already checks for the `(payload.type === 'span' || payload.type === 'root_span')` condition, seems like the assert here will always be true.

Reply:
synced up on slack, this assert is to resolve typescript complaints in following code since typescript doesn't actually recognized the `payload.type` from the boolean `hasCollapsedSpanGroup`.